tests/boot/boot_to_desktop.pm

```diff
@@ -42,4 +42,29 @@ sub test_flags {
     return {fatal => 1, milestone => 1};
 }
 
+sub post_fail_hook {
+    my $self = shift;
+
+    save_screenshot;
+
+    # check for text login to check if X has failed
+    if (check_screen('generic-login')) {
+        record_info 'Seems that the display manager failed';
+    }
+
+    # if we found a shell, we do not need the memory dump
+    if (!(match_has_tag('emergency-shell') or match_has_tag('emergency-mode'))) {
+        die "save_memory_dump not implemented, no way to save memory_dump" unless check_var('BACKEND', 'qemu');
+        diag 'Save memory dump to debug bootup problems, e.g. for bsc#1005313';
+        save_memory_dump;
+    }
```

Comment:
[DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself). You are calling the same code multiple times, please [extract method](http://refactoring.com/catalog/extractMethod.html).

Reply:
I fail to see a repetition?

Reply:
That section is copied from https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/fdafed9209eb3a1271316b8795d8c48560b7e549/tests/installation/first_boot.pm#L117

Reply:
The whole PFH is copied as suggested in the ticket...

Reply:
I don't see where the ticket suggests to copy the `post_fail_hook`. It says "Apply something comparable for non-s390x backends". Can you please try to not duplicate that? Or do you think we should duplicate that?

Reply:
I think its a good base for something more specific, both tests are intended for booting but lack a 'boot-basetest'
We can of course add such a basetest as next step